source/simulator/assembly.cc

```diff
@@ -1004,6 +1002,276 @@ namespace aspect
   }
 
 
+
+  /**
+   * A namespace for the definition of sub-namespaces in which we
+   * implement assemblers for the various terms in the linear systems
+   * ASPECT solves.
+   */
+  namespace Assemblers
+  {
+    /**
+     * A namespace for the definition of functions that implement the
+     * linear system terms for the *complete* compressible or
+     * incompressible equations.
+     */
+    namespace CompleteEquations
+    {
+      template <int dim>
+      void
+      local_assemble_stokes_preconditioner (const Introspection<dim>                                &introspection,
```

Comment:
Fair enough. My guiding line was to ask for as little as I could get away with, but a `SimulatorAccess` object makes things uniform. Will come up in the next patch iteration.